### PR TITLE
CPQ - Enable display price formula

### LINKED ISSUE #12414: CPQ Enable display price formula on configuration
Take into account that there are `ConfiguratorAttributeSingleSelectionBaseComponent` and `ConfiguratorAttributeMultiSelectionBaseComponent` base components that 'contain' `extractPriceFormulaParameters` method that prepares price formula.

**Wireframes:**
[M_CPS_Configuration_forms](https://4prsat.axshare.com/#id=jhn1qb&p=m_cps_configuation_forms&g=1)
[T_CPS_Configuration_forms](https://4prsat.axshare.com/#id=4vlcax&p=t_cps_configuration_forms&g=1)
[D_CPS_Configuration_forms](https://4prsat.axshare.com/#id=fr0uxa&p=d_cps_configuration_forms&g=1)
[W_CPS_Configuration_forms](https://4prsat.axshare.com/#id=zq2caa&p=w_cps_configuration_forms&g=1)

Price formula should be enabled for the following attribute types:
- [x] Drop-down list
- [x]  Radio buttons list
- [ ]  Input Field (not possible)
- [ ]  Image (only for VC / CPS)
- [x]  Checkbox list
- [x] Single checkbox behaves like checkbox list

**Done criteria:**
- [x] styling,
- [x] corresponding unit tests,
- [x] styling doc,
- [x] schematics (constructor)

### PR DESCRIPTION
**Pull request by Larisa-Staroverova**

### CODE REVIEW

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.spec.ts [hunk 1]
```diff
@@ -129,4 +148,47 @@ describe('ConfiguratorAttributeMultiSelectionBaseComponent', () => {
...
+        component.attribute.attributePriceTotal
+      );
+      expect(priceFormulaParameters?.isLightedUp).toBe(true);
+    });
+  });
+
+  describe('extractValuePriceFormulaParameters', () => {
+    it('should return `undefined`', () => {
+      expect(
+        component.extractValuePriceFormulaParameters(undefined)
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.spec.ts]: Strict mode issue

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { Configurator } from '../../../../core/model/configurator.model';
 import { ConfigFormUpdateEvent } from '../../../form/configurator-form.event';
 import { ConfiguratorAttributeQuantityComponentOptions } from '../../quantity/configurator-attribute-quantity.component';
 import { ConfiguratorAttributeQuantityService } from '../../quantity/configurator-attribute-quantity.service';
+import { ConfiguratorPriceComponentOptions } from '../../../price/configurator-price.component';
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts]: It seems your IDE does not sort the imports, could you enable that?

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts [hunk 3]
```diff
@@ -99,4 +100,40 @@ export abstract class ConfiguratorAttributeMultiSelectionBaseComponent extends C
 
     this.selectionChange.emit(event);
   }
+
+  /**
+   * Extract corresponding price formula parameters
+   *
+   * @return {ConfiguratorPriceComponentOptions} - New price formula
+   */
+  extractPriceFormulaParameters(): ConfiguratorPriceComponentOptions {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts:110]: Could you explain in JS doc why quantity and price are 0?

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts [hunk 4]
```diff
@@ -99,4 +100,40 @@ export abstract class ConfiguratorAttributeMultiSelectionBaseComponent extends C
...
+    return {
+      quantity: 0,
+      price: {
+        value: 0,
+        currencyIso: '',
+      },
+      priceTotal: this.attribute?.attributePriceTotal,
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts]:
```suggestion
      priceTotal: this.attribute.attributePriceTotal,
```
Attribute is mandatory

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts [hunk 5]
```diff
@@ -99,4 +100,40 @@ export abstract class ConfiguratorAttributeMultiSelectionBaseComponent extends C
...
+  /**
+   * Extract corresponding value price formula parameters
+   *
+   * @param {Configurator.Value} value - Configurator value
+   * @return {ConfiguratorPriceComponentOptions} - New price formula
+   */
+  extractValuePriceFormulaParameters(
+    value: Configurator.Value
+  ): ConfiguratorPriceComponentOptions | undefined {
+    if (value) {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts]: Not needed, as value is mandatory!

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts [hunk 6]
```diff
@@ -99,4 +100,40 @@ export abstract class ConfiguratorAttributeMultiSelectionBaseComponent extends C
...
+      return {
+        quantity: value?.quantity,
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts]:
```suggestion
        quantity: value.quantity,
```

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts [hunk 7]
```diff
@@ -99,4 +100,40 @@ export abstract class ConfiguratorAttributeMultiSelectionBaseComponent extends C
...
+        price: value?.valuePrice,
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts]: The same for that attribute and the following 2

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.spec.ts [hunk 8]
```diff
@@ -291,6 +291,25 @@ describe('ConfiguratorAttributeSingleSelectionBaseComponent', () => {
...
+        component.extractValuePriceFormulaParameters(undefined)
+      ).toBeUndefined();
+    });
+
+    it('should return price formula parameters', () => {
+      const value = createTestValue(100, 100, true);
+      const priceFormulaParameters = component.extractValuePriceFormulaParameters(
+        value
+      );
+      expect(priceFormulaParameters.price?.value).toBe(
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.spec.ts]:
```suggestion
      expect(priceFormulaParameters?.price?.value).toBe(
```
Strict mode issue

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.spec.ts [hunk 9]
```diff
@@ -291,6 +291,25 @@ describe('ConfiguratorAttributeSingleSelectionBaseComponent', () => {
...
+        value?.valuePrice?.value
+      );
+      expect(priceFormulaParameters.isLightedUp).toBe(value?.selected);
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.spec.ts]:
```suggestion
      expect(priceFormulaParameters?.isLightedUp).toBe(value?.selected);
```
strict

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.ts [hunk 10]
```diff
@@ -137,6 +137,23 @@ export abstract class ConfiguratorAttributeSingleSelectionBaseComponent extends
...
+   *
+   * @param {Configurator.Value} value - Configurator value
+   * @return {ConfiguratorPriceComponentOptions} - New price formula
+   */
+  extractValuePriceFormulaParameters(
+    value?: Configurator.Value
+  ): ConfiguratorPriceComponentOptions | undefined {
+    if (value) {
+      return {
+        price: value?.valuePrice,
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.ts]:
```suggestion
        price: value.valuePrice,
```
value is defined!

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.ts [hunk 11]
```diff
@@ -137,6 +137,23 @@ export abstract class ConfiguratorAttributeSingleSelectionBaseComponent extends
...
+        isLightedUp: value?.selected,
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.ts]:
```suggestion
        isLightedUp: value.selected,
```

File: feature-libs/product-configurator/rulebased/components/attribute/types/checkbox-list/configurator-attribute-checkbox-list.component.spec.ts [hunk 12]
```diff
@@ -208,33 +215,86 @@ describe('ConfigAttributeCheckBoxListComponent', () => {
...
+          value: 250,
+        },
+        valuePriceTotal: {
+          currencyIso: '$',
+          formattedValue: '$500.0',
+          value: 500,
+        },
+      };
+
+      component.attribute.values[0] = value;
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/types/checkbox-list/configurator-attribute-checkbox-list.component.spec.ts]: strict mode!

File: feature-libs/product-configurator/rulebased/components/attribute/types/drop-down/configurator-attribute-drop-down.component.spec.ts [hunk 13]
```diff
@@ -113,4 +125,75 @@ describe('ConfigAttributeDropDownComponent', () => {
...
+    });
+
+    it('should display quantity and price', () => {
+      component.attribute.quantity = 5;
+      component.attribute.attributePriceTotal = {
+        currencyIso: '$',
+        formattedValue: '500.00$',
+        value: 500,
+      };
+      component.attribute.values[0].valuePrice = {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/types/drop-down/configurator-attribute-drop-down.component.spec.ts]: strict mode

File: feature-libs/product-configurator/rulebased/components/attribute/types/drop-down/configurator-attribute-drop-down.component.spec.ts [hunk 14]
```diff
@@ -113,4 +125,75 @@ describe('ConfigAttributeDropDownComponent', () => {
...
+      fixture.detectChanges();
+      CommonConfiguratorTestUtilsService.expectElementNotPresent(
+        expect,
+        htmlElem,
+        'cx-configurator-attribute-quantity'
+      );
+    });
+
+    it('should display price formula', () => {
+      component.attribute.values[0].valuePrice = {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/types/drop-down/configurator-attribute-drop-down.component.spec.ts]: strict mode

File: feature-libs/product-configurator/rulebased/components/attribute/types/drop-down/configurator-attribute-drop-down.component.ts [hunk 15]
```diff
@@ -26,4 +27,8 @@ export class ConfiguratorAttributeDropDownComponent
   ngOnInit() {
     this.attributeDropDownForm.setValue(this.attribute?.selectedSingleValue);
   }
+
+  getSelectedValue(): Configurator.Value | undefined {
+    return this.attribute?.values?.find((value) => value?.selected);
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/types/drop-down/configurator-attribute-drop-down.component.ts]:
```suggestion
    return this.attribute.values?.find((value) => value?.selected);
```
attribute is mandatory

File: feature-libs/product-configurator/rulebased/components/attribute/types/radio-button/configurator-attribute-radio-button.component.spec.ts [hunk 16]
```diff
@@ -96,4 +124,75 @@ describe('ConfigAttributeRadioButtonComponent', () => {
...
+    });
+
+    it('should display quantity and price', () => {
+      component.attribute.quantity = 5;
+      component.attribute.attributePriceTotal = {
+        currencyIso: '$',
+        formattedValue: '500.00$',
+        value: 500,
+      };
+      component.attribute.values[0].valuePrice = {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/types/radio-button/configurator-attribute-radio-button.component.spec.ts]: strict mode issue

File: feature-libs/product-configurator/rulebased/components/attribute/types/radio-button/configurator-attribute-radio-button.component.spec.ts [hunk 17]
```diff
@@ -96,4 +124,75 @@ describe('ConfigAttributeRadioButtonComponent', () => {
...
+      fixture.detectChanges();
+      CommonConfiguratorTestUtilsService.expectElementNotPresent(
+        expect,
+        htmlElem,
+        'cx-configurator-attribute-quantity'
+      );
+    });
+
+    it('should display price formula', () => {
+      component.attribute.values[0].valuePrice = {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/types/radio-button/configurator-attribute-radio-button.component.spec.ts]: strict mode issue

File: feature-libs/product-configurator/rulebased/components/overview-attribute/configurator-overview-attribute.component.ts [hunk 18]
```diff
@@ -8,4 +11,24 @@ import { Configurator } from '../../core/model/configurator.model';
 })
 export class ConfiguratorOverviewAttributeComponent {
   @Input() attributeOverview: Configurator.AttributeOverview;
+
+  constructor(protected breakpointService: BreakpointService) {}
+
+  extractPriceFormulaParameters(): ConfiguratorPriceComponentOptions {
+    return {
+      quantity: this.attributeOverview?.quantity,
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-attribute/configurator-overview-attribute.component.ts]:
```suggestion
      quantity: this.attributeOverview.quantity,
```

File: feature-libs/product-configurator/rulebased/components/overview-attribute/configurator-overview-attribute.component.ts [hunk 19]
```diff
@@ -8,4 +11,24 @@ import { Configurator } from '../../core/model/configurator.model';
...
+      price: this.attributeOverview?.valuePrice,
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-attribute/configurator-overview-attribute.component.ts]:
```suggestion
      price: this.attributeOverview.valuePrice,
```
attributeOv is mandatory

File: feature-libs/product-configurator/rulebased/components/overview-attribute/configurator-overview-attribute.component.ts [hunk 20]
```diff
@@ -8,4 +11,24 @@ import { Configurator } from '../../core/model/configurator.model';
...
+      priceTotal: this.attributeOverview?.valuePriceTotal,
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-attribute/configurator-overview-attribute.component.ts]:
```suggestion
      priceTotal: this.attributeOverview.valuePriceTotal,
```

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts (hunk 5 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts (hunk 6 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts]: For VC / CPS quatity is not definied.

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts (hunk 7 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts]: For VC / CPS valuePriceTotal is not definied

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.spec.ts (hunk 8 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.spec.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.spec.ts (hunk 9 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.spec.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.ts (hunk 10 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.ts]: For VC / CPS `valuePrice `is not definied

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.ts (hunk 11 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/attribute/types/drop-down/configurator-attribute-drop-down.component.ts (hunk 15 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/drop-down/configurator-attribute-drop-down.component.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/overview-attribute/configurator-overview-attribute.component.ts (hunk 18 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/overview-attribute/configurator-overview-attribute.component.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/overview-attribute/configurator-overview-attribute.component.ts (hunk 19 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/overview-attribute/configurator-overview-attribute.component.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/overview-attribute/configurator-overview-attribute.component.ts (hunk 20 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/overview-attribute/configurator-overview-attribute.component.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/attribute/types/radio-button/configurator-attribute-radio-button.component.spec.ts (hunk 17 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/radio-button/configurator-attribute-radio-button.component.spec.ts]: By changing to `component.attribute.values[0]?.valuePrice` comes the following issue:
`The left-hand side of an assignment expression may not be an optional property access.`

File: feature-libs/product-configurator/rulebased/components/attribute/types/radio-button/configurator-attribute-radio-button.component.spec.ts (hunk 16 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/radio-button/configurator-attribute-radio-button.component.spec.ts]: By changing to `component.attribute.values[0]?.valuePrice` comes the following issue:
`The left-hand side of an assignment expression may not be an optional property access.`

File: feature-libs/product-configurator/rulebased/components/attribute/types/drop-down/configurator-attribute-drop-down.component.spec.ts (hunk 14 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/drop-down/configurator-attribute-drop-down.component.spec.ts]: By changing to `component.attribute.values[0]?.valuePrice` comes the following issue:
`The left-hand side of an assignment expression may not be an optional property access.`

File: feature-libs/product-configurator/rulebased/components/attribute/types/drop-down/configurator-attribute-drop-down.component.spec.ts (hunk 13 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/drop-down/configurator-attribute-drop-down.component.spec.ts]: By changing to `component.attribute.values[0]?.valuePrice` comes the following issue:
`The left-hand side of an assignment expression may not be an optional property access.`

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.spec.ts (hunk 1 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.spec.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts (hunk 4 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts]: `attribute` is mandatory but `attributePriceTotal` is not mandatory
the question mark is for `attributePriceTotal`
For VC / CPS `attributePriceTotal` is not defined

Comment (ChristophHi) [reply feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts]: My point is:` this.attribute.attributePriceTotal `is fine. No need to add '?', as this refers to this.attribute.
I know that attributePriceTotal is optional.

File: feature-libs/product-configurator/rulebased/components/attribute/types/radio-button/configurator-attribute-radio-button.component.spec.ts (hunk 16 above)
Comment (ChristophHi) [reply feature-libs/product-configurator/rulebased/components/attribute/types/radio-button/configurator-attribute-radio-button.component.spec.ts]: I only wanted to stress that this line causes a strict mode issue. To fix it, we need to check if the value we want to assign the valuePrice to is defined.

File: feature-libs/product-configurator/rulebased/components/attribute/types/radio-button/configurator-attribute-radio-button.component.spec.ts (hunk 17 above)
Comment (ChristophHi) [reply feature-libs/product-configurator/rulebased/components/attribute/types/radio-button/configurator-attribute-radio-button.component.spec.ts]: See my remark above

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts (hunk 6 above)
Comment (ChristophHi) [reply feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts]: Yes, I know. Still the chaining operator is not needed

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.ts (hunk 10 above)
Comment (ChristophHi) [reply feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.ts]: Yes, I know. Still the chaining operator is not needed

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts (hunk 2 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts (hunk 4 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.ts (hunk 10 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/attribute/types/checkbox-list/configurator-attribute-checkbox-list.component.spec.ts (hunk 12 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/checkbox-list/configurator-attribute-checkbox-list.component.spec.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/attribute/types/drop-down/configurator-attribute-drop-down.component.spec.ts (hunk 13 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/drop-down/configurator-attribute-drop-down.component.spec.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/attribute/types/radio-button/configurator-attribute-radio-button.component.spec.ts (hunk 16 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/radio-button/configurator-attribute-radio-button.component.spec.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.spec.ts [hunk 21]
```diff
@@ -291,6 +291,25 @@ describe('ConfiguratorAttributeSingleSelectionBaseComponent', () => {
...
+        component.extractValuePriceFormulaParameters(undefined)
+      ).toBeUndefined();
+    });
+
+    it('should return price formula parameters', () => {
+      const value = createTestValue(100, 100, true);
+      const priceFormulaParameters = component.extractValuePriceFormulaParameters(
+        value
+      );
+      expect(priceFormulaParameters?.price.value).toBe(
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.spec.ts]:
```suggestion
      expect(priceFormulaParameters?.price?.value).toBe(
```

Comment (ChristophHi) [reply feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.spec.ts]: Strict mode

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.spec.ts (hunk 21 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.spec.ts]: Done!

**Review by ChristophHi — DISMISSED** [review]

**Review by Platonn — APPROVED** [review]

**Pull request merged**